frame/multisig/src/lib.rs

```diff
@@ -435,20 +374,18 @@ decl_module! {
 		///   deposit taken for its lifetime of
 		///   `DepositBase + threshold * DepositFactor`.
 		/// ----------------------------------
-		/// - Base Weight:
-		///     - Create: 44.71 + 0.088 * S
-		///     - Approve: 31.48 + 0.116 * S
 		/// - DB Weight:
 		///     - Read: Multisig Storage, [Caller Account]
 		///     - Write: Multisig Storage, [Caller Account]
 		/// # </weight>
-		#[weight = weight_of::as_multi::<T>(
-			other_signatories.len(),
-			0, // call_len is zero in this case
-			*max_weight,
-			true, // assume worst case: calls write
-			true, // assume worst case: refunded
-		)]
+		#[weight = {
+			let s = other_signatories.len() as u32;
+
+			T::WeightInfo::approve_as_multi_create(s)
+			.max(T::WeightInfo::approve_as_multi_approve(s))
+			.max(T::WeightInfo::approve_as_multi_complete(s))
+			.saturating_add(*max_weight)
+		}]
```

Comment:
here the call is still decoded in some situation (like if it gets complete), thus decoding the code is still `O(call_len)` but anyway it should be small (like we don't hash it here). And it hasn't been introduced by this PR.